### PR TITLE
ER-1372 remap foundation and masters

### PR DESCRIPTION
**Pull request by mrpmorris**

### CODE REVIEW

File: src/Shared/Recruit.Shared.Web/Extensions/EnumExtensions.cs [hunk 1]
```diff
@@ -19,8 +19,10 @@ public static string GetDisplayName(this Enum enumValue)
 
         private static readonly Dictionary<Enum, string> DisplayNames = new Dictionary<Enum, string>
         {
+#pragma warning disable CS0618 // Type or member is obsolete
             { ProgrammeLevel.FoundationDegree, "Foundation Degree" },
             { ProgrammeLevel.Masters, "Master's Degree" },
```
Comment (sachpatel) [inline src/Shared/Recruit.Shared.Web/Extensions/EnumExtensions.cs]: Why can't we remove this if nothing is going to be set to this? Or is this after your first round of changes?

File: src/Shared/Recruit.Vacancies.Client/Domain/Entities/ProgrammeLevelHelper.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace Esfa.Recruit.Vacancies.Client.Domain.Entities
+{
+    public static class ProgrammeLevelHelper
+    {
+        public static bool TryRemapFromInt(int value, out ProgrammeLevel result)
```
Comment (sachpatel) [inline src/Shared/Recruit.Vacancies.Client/Domain/Entities/ProgrammeLevelHelper.cs]: Can this method please be static as it is only being called from one place which is within this class.

File: src/Shared/Recruit.Vacancies.Client/Domain/Entities/ProgrammeLevelHelper.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            switch (value)
+            {
+                case 5:
```
Comment (sachpatel) [inline src/Shared/Recruit.Vacancies.Client/Domain/Entities/ProgrammeLevelHelper.cs]: I assume this is more permanent as long as the rule exists to make the modification. Reading this class and understanding the particular change from x to y is difficult. Could you please put `case 5: // PhD` or whatever 5 was supposed to be and the same for `case 7:` please.

File: src/Shared/UnitTests/Vacancies.Client/Domain/Entities/ProgrammeLevelHelperTests.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            }
+        }
+
+        [Fact]
+        public void RemapFromInt_ShouldThrowArgumentException_WhenPassedAnIntWithNoCorrespondingValue()
+        {
+            int intValueToConvert = (int)ProgrammeLevel.Degree + 2;
+            Assert.Throws<ArgumentException>(() =>
+            {
+                ProgrammeLevel result = ProgrammeLevelHelper.RemapFromInt(intValueToConvert);
```
Comment (sachpatel) [inline src/Shared/UnitTests/Vacancies.Client/Domain/Entities/ProgrammeLevelHelperTests.cs]: you don't need the `ProgrammeLevel result =` do you?

File: src/Shared/UnitTests/Vacancies.Client/Domain/Entities/ProgrammeLevelHelperTests.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            });
+        }
+
+        [Fact]
+        public void TryRemapFromInt_ShouldSucceedAndReturnHigher_WhenConvertingFromFoundation5()
+        {
+            bool success = ProgrammeLevelHelper.TryRemapFromInt(5, out ProgrammeLevel result);
+            success.Should().Be(true);
```
Comment (sachpatel) [inline src/Shared/UnitTests/Vacancies.Client/Domain/Entities/ProgrammeLevelHelperTests.cs]: There are method `BeTrue()` and `BeFalse()` if that helps.

File: src/Shared/Recruit.Vacancies.Client/Domain/Entities/ProgrammeLevelHelper.cs (hunk 2 above)
Comment (sachpatel) [reply src/Shared/Recruit.Vacancies.Client/Domain/Entities/ProgrammeLevelHelper.cs]: Sorry meant `private static`

File: src/Shared/Recruit.Shared.Web/Extensions/EnumExtensions.cs (hunk 1 above)
Comment (mrpmorris) [reply src/Shared/Recruit.Shared.Web/Extensions/EnumExtensions.cs]: I'm disabling compiler warnings for a known violation. I don't like compiler warnings, ideally we should treat warnings as errors.

Comment (sachpatel) [reply src/Shared/Recruit.Shared.Web/Extensions/EnumExtensions.cs]: You can put in treat warnings as errors, so long as you put it across all projects

File: src/Shared/Recruit.Vacancies.Client/Domain/Entities/ProgrammeLevelHelper.cs (hunk 2 above)
Comment (mrpmorris) [reply src/Shared/Recruit.Vacancies.Client/Domain/Entities/ProgrammeLevelHelper.cs]: I have some tests for it, so I'll leave it public

File: src/Shared/Recruit.Shared.Web/Extensions/EnumExtensions.cs (hunk 1 above)
Comment (mrpmorris) [reply src/Shared/Recruit.Shared.Web/Extensions/EnumExtensions.cs]: I know, and we haven't done that, but that's no reason to allow warnings if they can be avoided.

Comment (sachpatel) [reply src/Shared/Recruit.Shared.Web/Extensions/EnumExtensions.cs]: Which is the reason to dump this code, particularly in this class where it's for a UI concern.

**Pull request closed without merging**